storagemarket/deal_execution.go

```diff
@@ -81,10 +77,13 @@ func (p *Provider) execDeal(ctx context.Context, pub event.Emitter, deal *types.
 
 	// Transfer Data
 	if deal.Checkpoint < dealcheckpoints.Transferred {
-		if err := p.transferAndVerify(ctx, pub, deal); err != nil {
+		if err := p.transferAndVerify(dh.transferCtx, pub, deal); err != nil {
+			dh.transferCancelled(nil)
```

Comment:
The transfer is not actually cancelled here right? Maybe this method should be called `transferCompleted`

Reply:
This method call is to inform the deal handler that a transfer isn't going to go through. What we are saying here is that the transfer not going through is successful so that the deal handler can return a nil error in response to a transfer cancellation call. Open to suggestions here.

Reply:
Ok